### PR TITLE
[DLLS] Export improvements

### PR DESCRIPTION
**Pull request by tkreuzer**
## Purpose

Improve some export definitions.

## Proposed changes

- [ADVAPI32][ETWTRACE] Add etwtrace library and link advapi32 to it on NT6+
- [SOFTPUB] Replace some stubs with forwarders
- [WINSPOOL.DRV] Improve some stubs
- [MSVCRT] Export _vcwprintf

## Tests

- ✅ KVM x86: https://reactos.org/testman/compare.php?ids=98291,98295,98300,98303
- ✅ KVM x64: https://reactos.org/testman/compare.php?ids=98355,98358,98361,98362

### CODE REVIEW

File: dll/win32/advapi32/advapi32.spec [hunk 1]
```diff
@@ -220,17 +220,17 @@
 @ stdcall EnumServicesStatusExW(long long long long ptr long ptr ptr ptr wstr)
 @ stdcall EnumServicesStatusW(long long long ptr long ptr ptr ptr)
 @ stdcall -version=0x502 EnumerateTraceGuids(ptr long ptr) ntdll.EtwEnumerateTraceGuids
-@ stdcall -stub -version=0x600+ EnumerateTraceGuids(ptr long ptr)
+@ stdcall -stub -version=0x600+ EnumerateTraceGuids(ptr long ptr) # EtwEnumerateTraceGuids
```
Comment (whindsaks) [inline dll/win32/advapi32/advapi32.spec:223]: Is this commented out on purpose? Same question further down...

File: dll/ntdll/CMakeLists.txt [hunk 2]
```diff
@@ -78,3 +81,6 @@ add_cd_file(TARGET ntdll DESTINATION reactos/system32 NO_CAB FOR all)
 if(MSVC AND ARCH STREQUAL "i386")
     target_link_libraries(libntdll ftol2)
 endif()
+
+if(DLL_EXPORT_VERSION GREATER_EQUAL 0x600)
```
Comment (learn-more) [inline dll/ntdll/CMakeLists.txt:85]: Refactoring leftover?

File: dll/win32/advapi32/advapi32.spec (hunk 1 above)
Comment (tkreuzer) [reply dll/win32/advapi32/advapi32.spec:223]: No, on NT6+ it's not a forwarder, it's implemented in advapi32. The comment is just a hint regarding the different name. It should eventually be implemented in the eetwtrace lib, I was just too lazy to add stubs there.

**Review by learn-more — APPROVED** [review]

File: dll/ntdll/CMakeLists.txt [hunk 3]
```diff
@@ -24,14 +24,18 @@ add_library(rtl_um OBJECT
 target_link_libraries(rtl_um apisets ${PSEH_LIB})
 add_dependencies(rtl_um psdk)
 
+# On NT6+ this is used by advapi32
```
Comment (HBelusca) [inline dll/ntdll/CMakeLists.txt:27]: :+1:

File: dll/win32/advapi32/CMakeLists.txt [hunk 4]
```diff
@@ -66,3 +66,7 @@ add_delay_importlibs(advapi32 secur32)
 add_importlibs(advapi32 advapi32_vista rpcrt4 kernel32 ntdll)
 add_pch(advapi32 advapi32.h "${PCH_SKIP_SOURCE}")
 add_cd_file(TARGET advapi32 DESTINATION reactos/system32 FOR all)
+
+if(DLL_EXPORT_VERSION GREATER_EQUAL 0x600)
+    target_link_libraries(advapi32 etwtrace)
```
Comment (HBelusca) [inline dll/win32/advapi32/CMakeLists.txt:71]: For "locality" i would suggest you move it just below the other target_link_libraries that is above.

File: dll/win32/advapi32/advapi32.spec [hunk 5]
```diff
@@ -654,7 +654,7 @@
 @ stdcall SystemFunction040(ptr long long) # RtlEncryptMemory
 @ stdcall SystemFunction041(ptr long long) # RtlDecryptMemory
 @ stdcall -version=0x502 TraceEvent(double ptr) ntdll.EtwTraceEvent
-@ stdcall -stub -version=0x600+ TraceEvent(double ptr)
+@ stdcall -version=0x600+ TraceEvent(double ptr) EtwTraceEvent
 @ stdcall TraceEventInstance(double ptr ptr ptr) ntdll.EtwTraceEventInstance
```
Comment (HBelusca) [inline dll/win32/advapi32/advapi32.spec:658]: These (Etw)TraceEventInstance, and the TraceMessage ones from below, stay in ntdll even on Vista+ ? (and the redirection still needed then?)

**Review by HBelusca — APPROVED** [review]

**Review by DarkFire01 — APPROVED** [review]

File: dll/win32/advapi32/advapi32.spec (hunk 5 above)
Comment (tkreuzer) [reply dll/win32/advapi32/advapi32.spec:664]: yes

**Pull request merged**